### PR TITLE
434 Ban Command for account banishment

### LINKED ISSUE #434: [Command] /ban - banimento de um jogador

### PR DESCRIPTION
**Pull request by MarcusSilvaDev**

### CODE REVIEW

File: src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs [hunk 1]
```diff
@@ -71,6 +71,18 @@ public void Configure(EntityTypeBuilder<AccountModel> builder)
...
 
+        builder.Property(e => e.BanishedAt)
+            .HasColumnName("banishedAt")
+            .HasColumnType("datetime");
+        
+        builder.Property(e => e.BanishedReason)
+            .HasColumnName("banishedReason")
+            .HasColumnType("varchar(255)");
+        
+        builder.Property(e => e.AccountThatBanned)
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs]: maybe a better name could be BannedBy

File: src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs [hunk 2]
```diff
@@ -85,7 +97,19 @@ private static void Seed(EntityTypeBuilder<AccountModel> builder)
                 Email = "god@gmail.com",
                 Password = "1",
                 PremiumTime = 30,
-                AllowManyOnline = true
+                AllowManyOnline = true,
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs:100]: can you remove that model seed. I think we don't need that in the codebase

File: src/Database/NeoServer.Data/Configurations/ForSqLite/ForSQLiteAccountModelConfiguration.cs [hunk 3]
```diff
@@ -85,8 +98,19 @@ private static void Seed(EntityTypeBuilder<AccountModel> builder)
                 Password = "1",
                 PremiumTime = 30,
                 AllowManyOnline = true
+            },
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Configurations/ForSqLite/ForSQLiteAccountModelConfiguration.cs]: same

File: src/Database/NeoServer.Data/Model/AccountModel.cs [hunk 4]
```diff
@@ -22,6 +22,10 @@ public AccountModel()
     public string Secret { get; set; }
     public bool AllowManyOnline { get; set; }
     public AccountType Type { get; set; }
+    
+    public DateTime? BanishedAt { get; set; }
+    public string BanishedReason { get; set; }
+    public uint? AccountThatBanned { get; set; }
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Model/AccountModel.cs]: BannedBy?

File: src/Database/NeoServer.Data/Repositories/AccountRepository.cs [hunk 5]
```diff
@@ -199,6 +199,18 @@ public async Task SavePlayerInventory(IPlayer player)
...
 
+    public async Task<int> Ban(uint playerAccountId, string reason, uint bannedByAccountId)
+    {
+        await using var context = NewDbContext;
+
+        return await context.Accounts
+            .Where(x => x.AccountId == playerAccountId)
+            .ExecuteUpdateAsync(x
+                => x.SetProperty(y => y.AccountThatBanned, bannedByAccountId)
+                    .SetProperty(y => y.BanishedReason, reason)
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Repositories/AccountRepository.cs]: BanishmentReason?

File: src/Database/NeoServer.Data/Seeds/PlayerModelSeed.cs [hunk 6]
```diff
@@ -93,7 +93,7 @@ public static void Seed(EntityTypeBuilder<PlayerModel> builder)
             {
                 PlayerId = 3,
                 PlayerType = 1,
-                AccountId = 1,
+                AccountId = 2,
```
Comment (caioavidal) [inline src/Database/NeoServer.Data/Seeds/PlayerModelSeed.cs]: please remove the change on the seed

File: src/NetworkingServer/NeoServer.Networking.Handlers/LogIn/AccountLoginHandler.cs [hunk 7]
```diff
@@ -53,6 +53,12 @@ public override async void HandleMessage(IReadOnlyNetworkMessage message, IConne
             return;
         }
 
+        if (foundedAccount.BanishedAt is not null)
+        {
+            connection.Disconnect("Your account has been banished. reason: " + foundedAccount.BanishedReason);
```
Comment (caioavidal) [inline src/NetworkingServer/NeoServer.Networking.Handlers/LogIn/AccountLoginHandler.cs]: "Reason" instead of "reason"

**Review by caioavidal — CHANGES_REQUESTED** [review]

File: src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs (hunk 2 above)
Comment (MarcusSilvaDev) [reply src/Database/NeoServer.Data/Configurations/AccountModelConfiguration.cs:100]: Sure

**Review by caioavidal — APPROVED** [review]

**Pull request merged**